### PR TITLE
Accept all Platform Interface Manager platform names in ASE configuration

### PR DESCRIPTION
**Pull request by michael-adler**
- Add mapping from physical platform name to ASE platforms in platform db.
- afu_platform_config writes ASE platform name to a file.
- Add the ability to set the default physical platform name in ASE using
  an environment variable (OPAE_ASE_DEFAULT_PLATFORM).

### CODE REVIEW

**Comment by michael-adler** [comment]
Regressions pass: http://sw-pert.altera.com/pert/pert.php?test_run_id=2129694

This change fixes a hole in the modeling of physical platforms. There are multiple physical platforms with small variations (e.g. different number of memory banks) that all map to the same ASE platform name. The change shifts responsibility to the Platform Interface Manager to handle the mapping.

File: ase/scripts/generate_ase_environment.py [hunk 1]
```diff
@@ -429,6 +429,7 @@ def gen_afu_platform_ifc(json_file):
         cfg.append("--ifc=" + default_ifc)
 
     if (args.platform == 'discrete'):
+        # Compatibility mode for original "discrete" platform argument.
         cfg.append("discrete_pcie3")
```
Comment (deepakcu84) [inline ase/scripts/generate_ase_environment.py]: what is discrete_pcie3?

Comment (michael-adler) [reply ase/scripts/generate_ase_environment.py]: The name of the RC platform interface in the Platform Interface Manager.

See https://github.com/OPAE/opae-sdk/blob/master/platforms/platform_db/discrete_pcie3.json

File: ase/scripts/generate_ase_environment.py [hunk 2]
```diff
@@ -478,8 +505,9 @@ def gen_afu_json_verilog_macros(json_file):
                             will be parsed by rtl_src_config.""")
 parser.add_argument('-t', '--tool', choices=['VCS', 'QUESTA'], default=None,
                     help='simulator tool to use, default is VCS if present')
-parser.add_argument('-p', '--platform', choices=['intg_xeon', 'discrete'],
-                    default='intg_xeon', help='FPGA Platform to simulate')
+parser.add_argument('-p', '--platform',
+                    default=get_default_platform(),
+                    help='FPGA Platform to simulate')
 parser.add_argument('-x', '--exclude', default=None,
```
Comment (deepakcu84) [inline ase/scripts/generate_ase_environment.py]: can we add an option to list all supported platforms?

Comment (michael-adler) [reply ase/scripts/generate_ase_environment.py]: afu_platform_config --help

**Pull request merged**